src/Kiota.Builder/Extensions/OpenApiSchemaExtensions.cs

```diff
@@ -68,6 +72,12 @@ public static bool IsObject(this OpenApiSchema? schema)
     {
         return "object".Equals(schema?.Type, StringComparison.OrdinalIgnoreCase);
     }
+
+    public static bool IsScalar(this OpenApiSchema? schema)
```

Comment:
this method name is confusing. An array would technically return true even though semantically it's not scalar.

Also the implementation checks two separate concerns: the structure (object) and the location (reference) of the schema.